lib/samples/navigate_route_with_rerouting/navigate_route_with_rerouting.dart

```diff
@@ -0,0 +1,532 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'dart:io';
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/common/common.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_tts/flutter_tts.dart';
+import 'package:path_provider/path_provider.dart';
+
+class NavigateRouteWithRerouting extends StatefulWidget {
+  const NavigateRouteWithRerouting({super.key});
+
+  @override
+  State<NavigateRouteWithRerouting> createState() =>
+      _NavigateRouteWithReroutingState();
+}
+
+class _NavigateRouteWithReroutingState extends State<NavigateRouteWithRerouting>
```

Comment:
General comment - it's quite slow to load - one thing to consider, are there any of the initial `async` operations that aren't sequential and could be loaded alongside one another with a `Future.wait(future1, future2)` as opposed to `await future1; await future2`.

Reply:
I added another indicator to load data and initialize navigation. The UI performance is improved, give a try,

Reply:
The two downloads can be started earlier and run simultaneously.

In `initState()`, we could do:
```dart
  _geodatabaseFuture = downloadSanDiegoGeodatabase();
  _tourPathFuture = downloadSanDiegoTourPath();
```
Then,  where the Future return values are needed,
```dart
final geodatabasePath = await _geodatabaseFuture;
```
(and the same for the other Future where its value is needed)

Reply:
Actually, instead of `_tourPathFuture`, you could have:
```dart
_locationDataSourceFuture = getLocationDataSource();
```

Reply:
updated